listenbrainz/websockets/listens_dispatcher.py

```diff
@@ -1,81 +1,70 @@
 import json
-import pika
 import time
-import threading
+
+from kombu.mixins import ConsumerMixin
 
 from listenbrainz.utils import get_fallback_connection_name
 
-from flask import current_app
-from listenbrainz.webserver.views.api_tools import LISTEN_TYPE_PLAYING_NOW, LISTEN_TYPE_IMPORT
+from kombu import Connection, Exchange, Queue, Consumer
+from kombu.utils.debug import setup_logging
+
+setup_logging()
 
 
-class ListensDispatcher(threading.Thread):
+class ListensDispatcher(ConsumerMixin):
 
     def __init__(self, app, socketio):
-        threading.Thread.__init__(self)
         self.app = app
         self.socketio = socketio
+        self.connection = None
+        self.channel2 = None
```

Comment:
channel2 is not a great name -- what is it used for?

Reply:
Renamed to playing_now_channel.